crates/astria-utils/src/config.rs

```diff
@@ -0,0 +1,17 @@
+use figment::{
+    providers::Env as FigmentEnv,
+    Figment,
+};
+use serde::{
+    Deserialize,
+    Serialize,
+};
+
+pub trait AstriaConfig<'a>: Serialize + Deserialize<'a> {
+    fn from_environment(env_prefix: &str) -> Result<Self, figment::Error> {
```

Comment:
nit: for compatibility you can make this this return a custom `Error` with `figment::Error` being left as an implementation detail, i.e.

```rust
pub struct Error {
    inner: figment::Error,
}

impl Error for Error {
    fn source(&self) -> Option<&dyn std::error::Error + 'static> {
        Some(&self.inner)
    }
}
```

Reply:
Happy to open up a future issue for this one, but I don't believe this is urgent for functionality